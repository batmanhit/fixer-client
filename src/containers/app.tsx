import * as React from 'react';
import { FunctionComponent } from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { Button, Navbar, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';

import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-bootstrap.css';

import './app.css';

import List from './pages/list';
import Create from './pages/create';
import Edit from './pages/edit';

export const App: FunctionComponent<{}> = () => {
  return (
    <div className="App">
      <Navbar color="dark" expand="md">
        <NavbarBrand href="/" className="mr-auto text-white">FIXER TEST</NavbarBrand>
        <Nav className="ml-auto" navbar={true}>
          <NavItem>
            <NavLink href="/create">
              <Button color="primary">Create</Button>
            </NavLink>
          </NavItem>
        </Nav>
      </Navbar>
      <Switch>
				<Route exact={true} path="/" component={List} />
        <Route exact={true} path="/create" component={Create} />
        <Route exact={true} path="/:id" component={Edit} />
      </Switch>
    </div>
  );
};

export default withRouter(App);
