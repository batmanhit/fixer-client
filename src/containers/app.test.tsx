import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { NavLink } from 'reactstrap';

import { App } from './app';
// import { Route } from 'react-router';

describe('Whole Application', () => {

  const testElement: JSX.Element = <App/>;
  const wrapper: ShallowWrapper = shallow(testElement);

  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render NavLink correctly', () => {
    const navLink = wrapper.find(NavLink);
    expect(navLink.prop('href')).toEqual('/create');
  });
});