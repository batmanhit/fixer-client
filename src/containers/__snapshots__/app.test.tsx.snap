// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whole Application snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar
          color="dark"
          expand="md"
          tag="nav"
>
          <NavbarBrand
                    className="mr-auto text-white"
                    href="/"
                    tag="a"
          >
                    FIXER TEST
          </NavbarBrand>
          <Nav
                    className="ml-auto"
                    navbar={true}
                    tag="ul"
                    vertical={false}
          >
                    <NavItem
                              tag="li"
                    >
                              <NavLink
                                        href="/create"
                                        tag="a"
                              >
                                        <Button
                                                  color="primary"
                                                  tag="button"
                                        >
                                                  Create
                                        </Button>
                              </NavLink>
                    </NavItem>
          </Nav>
</Navbar>,
        <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/create"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/:id"
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavbarBrand
              className="mr-auto text-white"
              href="/"
              tag="a"
>
              FIXER TEST
</NavbarBrand>,
            <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
>
              <NavItem
                            tag="li"
              >
                            <NavLink
                                          href="/create"
                                          tag="a"
                            >
                                          <Button
                                                        color="primary"
                                                        tag="button"
                                          >
                                                        Create
                                          </Button>
                            </NavLink>
              </NavItem>
</Nav>,
          ],
          "color": "dark",
          "expand": "md",
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "FIXER TEST",
              "className": "mr-auto text-white",
              "href": "/",
              "tag": "a",
            },
            "ref": null,
            "rendered": "FIXER TEST",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <NavItem
                tag="li"
>
                <NavLink
                                href="/create"
                                tag="a"
                >
                                <Button
                                                color="primary"
                                                tag="button"
                                >
                                                Create
                                </Button>
                </NavLink>
</NavItem>,
              "className": "ml-auto",
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavLink
                  href="/create"
                  tag="a"
>
                  <Button
                                    color="primary"
                                    tag="button"
                  >
                                    Create
                  </Button>
</NavLink>,
                "tag": "li",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    color="primary"
                    tag="button"
>
                    Create
</Button>,
                  "href": "/create",
                  "tag": "a",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Create",
                    "color": "primary",
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": "Create",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/create"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/:id"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/create",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar
            color="dark"
            expand="md"
            tag="nav"
>
            <NavbarBrand
                        className="mr-auto text-white"
                        href="/"
                        tag="a"
            >
                        FIXER TEST
            </NavbarBrand>
            <Nav
                        className="ml-auto"
                        navbar={true}
                        tag="ul"
                        vertical={false}
            >
                        <NavItem
                                    tag="li"
                        >
                                    <NavLink
                                                href="/create"
                                                tag="a"
                                    >
                                                <Button
                                                            color="primary"
                                                            tag="button"
                                                >
                                                            Create
                                                </Button>
                                    </NavLink>
                        </NavItem>
            </Nav>
</Navbar>,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/create"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/:id"
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <NavbarBrand
                className="mr-auto text-white"
                href="/"
                tag="a"
>
                FIXER TEST
</NavbarBrand>,
              <Nav
                className="ml-auto"
                navbar={true}
                tag="ul"
                vertical={false}
>
                <NavItem
                                tag="li"
                >
                                <NavLink
                                                href="/create"
                                                tag="a"
                                >
                                                <Button
                                                                color="primary"
                                                                tag="button"
                                                >
                                                                Create
                                                </Button>
                                </NavLink>
                </NavItem>
</Nav>,
            ],
            "color": "dark",
            "expand": "md",
            "tag": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "FIXER TEST",
                "className": "mr-auto text-white",
                "href": "/",
                "tag": "a",
              },
              "ref": null,
              "rendered": "FIXER TEST",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavItem
                  tag="li"
>
                  <NavLink
                                    href="/create"
                                    tag="a"
                  >
                                    <Button
                                                      color="primary"
                                                      tag="button"
                                    >
                                                      Create
                                    </Button>
                  </NavLink>
</NavItem>,
                "className": "ml-auto",
                "navbar": true,
                "tag": "ul",
                "vertical": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    href="/create"
                    tag="a"
>
                    <Button
                                        color="primary"
                                        tag="button"
                    >
                                        Create
                    </Button>
</NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      color="primary"
                      tag="button"
>
                      Create
</Button>,
                    "href": "/create",
                    "tag": "a",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Create",
                      "color": "primary",
                      "tag": "button",
                    },
                    "ref": null,
                    "rendered": "Create",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/create"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/:id"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/create",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
